src/components/date/__internal__/utils.test.ts

```diff
@@ -0,0 +1,448 @@
+  (format) => {
+    expect(formattedValue(format, new Date("2022-01-01"))).toEqual(
+      result(format)
+    );
+  }
+);
+
+test("should return an empty string when no date passed to `formattedValue`", () => {
+  expect(formattedValue("ddMMyyyy")).toEqual("");
+});
+
+test("should return the passed format and value if years in format string passed to `additionalYears` is not of length 2", () => {
+  expect(additionalYears("dd/MM/yyyy", "01/01/2022")).toEqual([
+    "dd/MM/yyyy",
+    "01/01/2022",
+  ]);
+
+  expect(additionalYears("yyyy/MM/dd", "2022/01/01")).toEqual([
+    "yyyy/MM/dd",
+    "2022/01/01",
+  ]);
+});
+
+describe.each(yearValuesLessThan69)(
+  "when year is two characters long and denotes years between 2000 and 2068",
+  (year) => {
+    test.each([
+      ["d.M.yy", `01-01-${year}`, ["d.M.yyyy", `01-01-20${year}`]],
+      ["d,M,yy", `01-01-${year}`, ["d,M,yyyy", `01-01-20${year}`]],
+      ["d-M-yy", `01-01-${year}`, ["d-M-yyyy", `01-01-20${year}`]],
+      ["d/M/yy", `01-01-${year}`, ["d/M/yyyy", `01-01-20${year}`]],
+      ["dd.M.yy", `01-01-${year}`, ["dd.M.yyyy", `01-01-20${year}`]],
+      ["dd,M,yy", `01-01-${year}`, ["dd,M,yyyy", `01-01-20${year}`]],
+      ["dd-M-yy", `01-01-${year}`, ["dd-M-yyyy", `01-01-20${year}`]],
+      ["dd/M/yy", `01-01-${year}`, ["dd/M/yyyy", `01-01-20${year}`]],
+      ["ddMMyy", `01-01-${year}`, ["ddMMyyyy", `01-01-20${year}`]],
+      ["dd.MM.yy", `01-01-${year}`, ["dd.MM.yyyy", `01-01-20${year}`]],
+      ["dd,MM,yy", `01-01-${year}`, ["dd,MM,yyyy", `01-01-20${year}`]],
+      ["dd-MM-yy", `01-01-${year}`, ["dd-MM-yyyy", `01-01-20${year}`]],
+      ["dd/MM/yy", `01-01-${year}`, ["dd/MM/yyyy", `01-01-20${year}`]],
+      ["yy.M.d", `${year}-01-01`, ["yyyy.M.d", `20${year}-01-01`]],
+      ["yy,M,d", `${year}-01-01`, ["yyyy,M,d", `20${year}-01-01`]],
+      ["yy-M-d", `${year}-01-01`, ["yyyy-M-d", `20${year}-01-01`]],
+      ["yy/M/d", `${year}-01-01`, ["yyyy/M/d", `20${year}-01-01`]],
+      ["yy.M.dd", `${year}-01-01`, ["yyyy.M.dd", `20${year}-01-01`]],
+      ["yy,M,dd", `${year}-01-01`, ["yyyy,M,dd", `20${year}-01-01`]],
+      ["yy-M-dd-M", `${year}-01-01`, ["yyyy-M-dd-M", `20${year}-01-01`]],
+      ["yy/M/dd/M", `${year}-01-01`, ["yyyy/M/dd/M", `20${year}-01-01`]],
+      ["yyMdd", `${year}-01-01`, ["yyyyMdd", `20${year}-01-01`]],
+      ["yy.MM.dd", `${year}-01-01`, ["yyyy.MM.dd", `20${year}-01-01`]],
+      ["yy,MM,dd", `${year}-01-01`, ["yyyy,MM,dd", `20${year}-01-01`]],
+      ["yy-MM-dd", `${year}-01-01`, ["yyyy-MM-dd", `20${year}-01-01`]],
+      ["yy/MM/dd", `${year}-01-01`, ["yyyy/MM/dd", `20${year}-01-01`]],
+    ])(
+      "returns expected output when format is `%s` and reference date is `%s`",
+      (format, referenceDate, expected) => {
+        expect(additionalYears(format, referenceDate)).toEqual(expected);
+      }
+    );
+  }
+);
+
+describe.each(yearValuesGreaterThan69)(
```

Comment:
comment(non-blocking): same as before

Excessive looping through so many dates (more this time)